plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/TempFileWriter.java

```diff
@@ -76,7 +81,17 @@ private static OrcWriter createOrcFileWriter(OrcDataSink sink, List<Type> types)
                 sink,
                 columnNames,
                 types,
-                OrcType.createRootOrcType(columnNames, types),
+                OrcType.createRootOrcType(columnNames, types, Optional.of(type -> {
```

Comment:
This function has a `@VisibleForTesting` annotation that should be removed if you're calling it here.

Reply:
We could also probably reuse some code from the `TypeConverter` class

Reply:
i thought about that, but i think that would be hard due to column IDs. this would work if we were writing a fully proper Iceberg file and reading it back as an iceberg file, but `SortingFileWriter` doesn't know iceberg.

I think i would rather eventually go into opposite direction. as discussed internally, we could write those temp files using PagesSerde+LZ4. `SortingFileWriter` uses ORC files only for compression benefits (to save IO).
cc @raunaqmorarka @losipiuk @lukasz-stec